src/tests/class_tests/openms/source/MzMLFile_test.cpp

```diff
@@ -1194,13 +1197,47 @@ START_SECTION(void transform(const String& filename_in, Interfaces::IMSDataConsu
 }
 END_SECTION
 
+START_SECTION([EXTRA])
+{
+  // Lade ein MzML-Testfile
+  MSExperiment exp;
+  MzMLFile mzml;
+  mzml.load(OPENMS_GET_TEST_DATA_PATH("ChromatogramExtractor_input.mzML"), exp);
+
+  // Speichere es mit GZIP-Kompression über deine Boost-Integration
+  std::string compressed_file;
+  NEW_TMP_FILE_EXT(compressed_file, ".gz");
+  mzml.store(compressed_file, exp);
+
+  // Stelle sicher, dass Datei geschrieben wurde
+  TEST_EQUAL(File::exists(compressed_file), true);
+
+  // Lade erneut über OpenMS
+  MSExperiment exp2;
+  mzml.load(compressed_file, exp2);
+
+  // Validierung
+  TEST_EQUAL(exp.getNrSpectra(), exp2.getNrSpectra());
+  TEST_EQUAL(exp.getNrChromatograms(), exp2.getNrChromatograms());
+  for (Size s = 0; s < exp.size(); ++s)
+  {
+    TEST_EQUAL(exp[s].size(), exp2[s].size());
+    for (Size p = 0; p < exp[s].size(); ++p)
+    {
+      TEST_REAL_SIMILAR(exp[s][p].getMZ(), exp2[s][p].getMZ());
+      TEST_REAL_SIMILAR(exp[s][p].getIntensity(), exp2[s][p].getIntensity());
+    }
```

Comment:
nice!

Reply:
with a bit of luck, you can also write: `TEST_EQUAL(exp, exp2)` since https://github.com/OpenMS/OpenMS/blob/develop/src/openms/include/OpenMS/KERNEL/MSExperiment.h#L117

Reply:
this still holds.
use `TEST_EQUAL(exp, exp2)`